core/src/avm2/globals/flash/media/video.rs

```diff
@@ -1,5 +1,68 @@
-use crate::avm2::{Activation, Error, Object, TObject, Value};
-use crate::display_object::TDisplayObject;
+use crate::avm2::globals::flash::display::display_object::initialize_for_allocator;
+use crate::avm2::{Activation, ClassObject, Error, Object, TObject, Value};
+use crate::display_object::{TDisplayObject, Video};
+
+pub fn video_allocator<'gc>(
+    class: ClassObject<'gc>,
+    activation: &mut Activation<'_, 'gc>,
+) -> Result<Object<'gc>, Error<'gc>> {
+    let video_class = activation.avm2().classes().video;
+
+    let mut target_class = Some(class);
+    while let Some(target) = target_class {
+        if target == video_class {
+            let movie = activation.context.swf.clone();
+            let new_do = Video::new(activation.context.gc_context, movie, 0, 0, None);
+            return initialize_for_allocator(activation, new_do.into(), class);
+        }
+
+        if let Some((movie, symbol)) = activation
+            .context
+            .library
+            .avm2_class_registry()
+            .class_symbol(target)
+        {
+            let child = activation
+                .context
+                .library
+                .library_for_movie_mut(movie)
+                .instantiate_by_id(symbol, activation.context.gc_context)?;
+
+            return initialize_for_allocator(activation, child, class);
+        }
+
+        target_class = target.superclass_object();
+    }
+
+    unreachable!("A Video subclass should have Video in superclass chain");
+}
+
+/// Implements `flash.media.Video`'s `init` method, which is called from the constructor
+pub fn init<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    if let Some(this) = this {
+        if let Some(video) = this.as_display_object().and_then(|dobj| dobj.as_video()) {
+            let width = args
```

Comment:
I think it would be shorter to get `width` and `height` using `ParametersExt`.

Reply:
Good catch, I forgot `ParametersExt` existed in AVM2 land.